### DISCUSSION TITLE
HiFi-GAN training does not work.

### DISCUSSION THREAD

**Discussion by upskyy** [coqui-ai/TTS#1410] [General Q&A] [answered]
### To Reproduce

```bash
$ CUDA_VISIBLE_DEVICES=0 python3 TTS/recipes/ljspeech/hifigan/train_hifigan.py
```

```python
# train_hifigan.py
import os

from trainer import Trainer, TrainerArgs

from TTS.utils.audio import AudioProcessor
from TTS.vocoder.configs import HifiganConfig
from TTS.vocoder.datasets.preprocess import load_wav_data
from TTS.vocoder.models.gan import GAN

output_path = os.path.dirname(os.path.abspath(__file__))

config = HifiganConfig(
    batch_size=64,
    eval_batch_size=16,
    num_loader_workers=4,
    num_eval_loader_workers=4,
    run_eval=True,
    test_delay_epochs=5,
    epochs=1000,
    seq_len=8192,
    pad_short=2000,
    use_noise_augment=True,
    eval_split_size=10,
    print_step=25,
    print_eval=False,
    mixed_precision=False,
    lr_gen=1e-4,
    lr_disc=1e-4,
    data_path=os.path.join(output_path, "../LJSpeech-1.1/wavs/"),
    output_path=output_path,
)

# init audio processor
ap = AudioProcessor(**config.audio.to_dict())

# load training samples
eval_samples, train_samples = load_wav_data(config.data_path, config.eval_split_size)

# init model
model = GAN(config, ap)

# init the trainer and 🚀
trainer = Trainer(
    TrainerArgs(),
    config,
    output_path,
    model=model,
    train_samples=train_samples,
    eval_samples=eval_samples,
)
trainer.fit()
```

### Expected behavior

Even though it exceeds 200k, it still sounds like a mechanical sound. Do I need to edit the hyperparameters separately? (#1368)
I synthesized it with tacotron2 + HiFi-GAN.

<img width="1411" alt="스크린샷 2022-03-14 오전 11 21 23" src="https://user-images.githubusercontent.com/97871105/158093643-6efa95e6-b62a-49c3-8da6-1ebe54d542e9.png">

### Environment

- 🐸TTS Version (e.g., 1.3.0):  0.6.1
- PyTorch Version (e.g., 1.8) :  1.10.2
- Python version:  3.8.12
- OS (e.g., Linux):  Linux 
- CUDA/cuDNN version:  11.3.1
- GPU models and configuration: GeForce 3090
- How you installed PyTorch (`conda`, `pip`, source): conda
- Any other relevant information:

**Comment by dmazurok** [comment]
Hi! Did you manage to solve it? I face the same on my dataset.
![image](https://user-images.githubusercontent.com/33186111/165489220-6ddefa97-b714-4b19-83a3-2de5467993db.png)
****

**Comment by erogol** [comment, marked as answer]
Can you try the dev branch? We released a fix for GAN training.

**Reply by jhlfrfufyfn** [reply]
Does this fix covers a problem with occurring NaNs while training hifigan with mixed_precision:true ? Every time I try to train it like this, it keeps getting NaNs in few thousand steps

**Reply by erogol** [reply]
Dunno just try

**Comment by r7sa** [comment]
It seems to solve some problem for me. Training still in progress but pitch is already more natural.

**Reply by kksteini** [reply]
I'm a total newbie.
I was wondering about the pitch for training one's own HiFi-GAN, though perhaps I'm just misunderstanding vocoders :grin:.
It was only monotone for me.
How long did you train it for?
How much data did you train it on?

I tried with 30 minutes of audio as a test. Does pitch present itself with more data?

**Comment by ikm565** [comment]
Please help take a look at this issue #2139. I think I ran into the same problem, when training hifi-gan with the interface here doesn't work, I tried training with hifi-gan's source code ([link](https://github.com/jik876/hifi-gan)), it works well with tacotron ([tacotron source code](https://github.com/NVIDIA/tacotron2)) but when I combine it with the mel model in this project, it still fails to produce normal speech. So I'm wondering if it's not that hifi-gan is bad, but that there is something wrong with the MEL model. Has anyone tried to train a MEL model based on the source code of the paper and generate speech together with the hifi-gan vocoder of this project?
